### PR TITLE
Changes for postgresql

### PR DESCRIPTION
**Pull request by jbirdjavi**

### CODE REVIEW

**Comment by jbirdjavi** [comment]
Brakeman update is in #44

File: .travis.yml [hunk 1]
```diff
@@ -7,16 +8,17 @@ cache:
   yarn: true
 services:
   - redis-server
-  - mysql
+  - postgresql
```
Comment (knutsenm) [inline .travis.yml:11]: I don't think this is necessary if you load a newer version as an addon

File: .travis.yml [hunk 2]
```diff
@@ -7,16 +8,17 @@ cache:
...
+addons:
+  - postgresql: "13"
```
Comment (knutsenm) [inline .travis.yml:13]: https://docs.travis-ci.com/user/database-setup/#using-a-different-postgresql-version mentions the need for extra config in the case of v11 and v12, but doesn't say anything about v13.

File: db/schema.rb [hunk 3]
```diff
@@ -1,100 +1,99 @@
...
 
-  create_table "entity_transfers", force: :cascade, options: "ENGINE=InnoDB DEFAULT CHARSET=utf8 ROW_FORMAT=COMPACT" do |t|
-    t.integer "entity_id",    limit: 5,       null: false
-    t.integer "transfer_id",  limit: 5,       null: false
-    t.boolean "is_recipient", default: false, null: false
+  create_table "countries", id: false, force: :cascade do |t|
+    t.text "country"
+    t.text "code"
+    t.text "lat"
+    t.text "lon"
```
Comment (knutsenm) [inline db/schema.rb:24]: Where did that come from? I don't see it in the source.

File: db/schema.rb [hunk 4]
```diff
@@ -1,100 +1,99 @@
...
+    t.text "fax"
+    t.text "email"
+    t.text "dp_reg"
+    t.text "dp_contact"
+    t.text "signatory"
+    t.integer "is_eea", limit: 2, default: 0, null: false
+    t.text "country", null: false
+    t.text "area"
+    t.integer "is_areaoffice", limit: 2, default: 0, null: false
+    t.text "description"
```
Comment (knutsenm) [inline db/schema.rb:42]: Eep! I guess there never was a migration for these?

File: db/schema.rb [hunk 5]
```diff
@@ -1,100 +1,99 @@
...
+    t.string "legal_name", limit: 255, default: "", null: false
+    t.text "legal_other"
+    t.text "address"
+    t.text "telephone"
+    t.text "fax"
+    t.text "email"
+    t.text "dp_reg"
+    t.text "dp_contact"
+    t.text "signatory"
+    t.integer "is_eea", limit: 2, default: 0, null: false
```
Comment (knutsenm) [inline db/schema.rb:38]: Why not `t.boolean`? Same question for `is_areaoffice` and `entity_transfers.is_recipient`

File: db/schema.rb [hunk 6]
```diff
@@ -1,100 +1,99 @@
...
-    t.boolean "is_recipient", default: false, null: false
+  create_table "countries", id: false, force: :cascade do |t|
+    t.text "country"
+    t.text "code"
+    t.text "lat"
+    t.text "lon"
   end
 
-  add_index "entity_transfers", ["entity_id"], name: "fk_entities_transfers_entities_1", using: :btree
-  add_index "entity_transfers", ["transfer_id"], name: "fk_entities_transfers_transfers_1", using: :btree
```
Comment (knutsenm) [inline db/schema.rb:40]: Ten `add_index` were removed; a corresponding `t.index` is missing in some cases, like these. Perhaps for such a small app it doesn't matter?

File: db/schema.rb [hunk 7]
```diff
@@ -1,100 +1,99 @@
...
+    t.string "name", limit: 255, default: "", null: false
+    t.text "full_description", null: false
+    t.text "data_subjects", null: false
+    t.text "data_categories", null: false
+    t.text "data_recipients", null: false
+    t.text "additional_info"
+    t.text "storage_limit", null: false
+    t.text "data_sensitive", null: false
+    t.text "security_policy", null: false
+    t.text "context"
```
Comment (knutsenm) [inline db/schema.rb:67]: Where did that come from? I don't see it in the source. Same question for `transfer_types.context`

File: db/schema.rb [hunk 8]
```diff
@@ -1,100 +1,99 @@
...
+    t.index ["username"], name: "index_users_on_username_users", unique: true
   end
 
-  add_index "users", ["username"], name: "index_users_on_username", unique: true, using: :btree
-
-  add_foreign_key "entity_transfers", "entities", name: "fk_entities_transfers_entities_1"
-  add_foreign_key "entity_transfers", "transfers", name: "fk_entities_transfers_transfers_1"
-  add_foreign_key "purpose_transfers", "purposes", name: "fk_purposes_transfers_purposes_1"
-  add_foreign_key "purpose_transfers", "transfers", name: "fk_purposes_transfers_transfers_1"
-  add_foreign_key "transfers", "transfer_types", name: "fk_transfers_transfertype_1"
```
Comment (knutsenm) [inline db/schema.rb:99]: Any need for these?

**Review by knutsenm — CHANGES_REQUESTED** [review]

File: .travis.yml (hunk 2 above)
Comment (jbirdjavi) [reply .travis.yml:12]: Yeah the fact that it worked seemed good enough for me.

File: db/schema.rb (hunk 3 above)
Comment (jbirdjavi) [reply db/schema.rb:24]: I generated the schema.rb file based on the migration of data over from MySQL to Postgresql via AWS tools. The countries table did exist in the MySQL database, even if there's not reference to it in the previous schema.rb file.

File: db/schema.rb (hunk 6 above)
Comment (jbirdjavi) [reply db/schema.rb:40]: Hmmm, I'll have to look into that...  The AWS Schema Conversion Tool should have added everything.

Comment (jbirdjavi) [reply db/schema.rb:40]: Ahh apparently those failed to migrate over.  I figured it out and they're added now.

File: db/schema.rb (hunk 5 above)
Comment (jbirdjavi) [reply db/schema.rb:38]: It looks like my conversion tools are changing MySQL tinyint types to Postgresql smallint types.  I'll have to ask @soberstadt how he dealt with that.

File: db/schema.rb (hunk 4 above)
Comment (jbirdjavi) [reply db/schema.rb:42]: Apparently not...

File: db/schema.rb (hunk 7 above)
Comment (jbirdjavi) [reply db/schema.rb:67]: Good question, but they existed in MySQL.

File: db/schema.rb (hunk 8 above)
Comment (jbirdjavi) [reply db/schema.rb:99]: These are back in place now.

**Review by knutsenm — APPROVED** [review]

**Pull request merged**